load_balancer/src/services/load_balancer.rs

```diff
@@ -29,20 +29,38 @@ use crate::{
     },
 };
 
+struct ConnectionGuard {
+    host: SocketAddr,
+    state: Arc<RwLock<LoadBalancerState>>,
+}
+
+impl Drop for ConnectionGuard {
+    fn drop(&mut self) {
+        let state = self.state.clone();
+        let host = self.host;
+        
+        tokio::spawn(async move {
```

Comment:
I'm not sure how sane it is to spawn tasks in a drop implementation. I would create an asynchronous method to have more control over the task.